src/script/page/Modals/UserModal/UserModal.tsx

```diff
@@ -0,0 +1,184 @@
+import UserActions from 'Components/panel/UserActions';
+import {Config} from '../../../Config';
+import {User} from '../../../entity/User';
+import {noop} from 'Util/util';
+import {Core} from '../../../service/CoreSingleton';
+
+interface UserModalProps {
+  userId: QualifiedId;
+  userRepository: UserRepository;
+  actionsViewModel: ActionsViewModel;
+  onClose?: () => void;
+  userState?: UserState;
+  teamState?: TeamState;
+}
+
+const brandName = Config.getConfig().BRAND_NAME;
+
+const UserModalComponent: React.FC<UserModalProps> = ({
+  userId,
+  onClose = noop,
+  userRepository,
+  actionsViewModel,
+  userState = container.resolve(UserState),
+  teamState = container.resolve(TeamState),
+}) => {
+  const [isShown, setIsShown] = useState<boolean>(false);
+  const [userNotFound, setUserNotFound] = useState<boolean>(false);
+  const [user, setUser] = useState<User | null>(null);
+  const hide = () => setIsShown(false);
+  const onModalClosed = () => {
+    setUser(null);
+    setUserNotFound(false);
+    onClose();
+  };
+  const {classifiedDomains} = useKoSubscribableChildren(teamState, ['classifiedDomains']);
+  const {self, isActivatedAccount} = useKoSubscribableChildren(userState, ['self', 'isActivatedAccount']);
+  const {is_verified: isSelfVerified} = useKoSubscribableChildren(self, ['is_verified']);
+  const isFederated = container.resolve(Core).backendFeatures.isFederated;
+  const replaceLinkLegalHold = replaceLink(Config.getConfig().URL.SUPPORT.LEGAL_HOLD_BLOCK, '', 'read-more-legal-hold');
+
+  useEffect(() => {
+    if (userId) {
+      userRepository
+        .getUserById(userId)
+        .then(user => {
+          if (user.isDeleted) {
+            return setUserNotFound(true);
+          }
+          setUser(user);
+        })
+        .catch(() => setUserNotFound(true));
+      setIsShown(true);
+    }
+
+    return () => {
+      setUser(null);
+      setUserNotFound(false);
+    };
+  }, [userId.id, userId.domain]);
+
+  return (
+    <div className="user-modal">
+      <ModalComponent
+        isShown={isShown}
+        onBgClick={hide}
+        onClosed={onModalClosed}
+        data-uie-name={user ? 'modal-user-profile' : userNotFound ? 'modal-cannot-open-profile' : ''}
+      >
+        <div className="modal__header">
+          {userNotFound && (
+            <h2 className="modal__header__title" data-uie-name="status-modal-title">
+              {t('userNotFoundTitle', brandName)}
+            </h2>
+          )}
+          <Icon.Close className="modal__header__button" onClick={hide} data-uie-name="do-close" />
+        </div>
+        <div className={cx('modal__body user-modal__wrapper', {'user-modal__wrapper--max': !user && !userNotFound})}>
+          {user && (
+            <>
+              <UserDetails participant={user} isSelfVerified={isSelfVerified} classifiedDomains={classifiedDomains} />
+              <EnrichedFields user={user} showDomain={isFederated} />
+              {user.isBlockedLegalHold() && (
```

Comment:
Please use `useKoSubscirbableChildren` for all observable user properties. Else we are facing potential bugs

Reply:
https://github.com/wireapp/wire-webapp/pull/13187/commits/2fbd40ae0108723c10201a6a345f4c8d1a4147f2